dipy/reconst/dki.py

```diff
@@ -1671,32 +1675,59 @@ def __init__(self, gtab, fit_method="WLS", return_S0_hat=False,
             self.sdp = PositiveDefiniteLeastSquares(22, A=self.sdp_constraints)
 
         self.weights = fit_method in {'WLS', 'WLLS', 'UWLLS', 'CWLS'}
+
         self.is_multi_method = fit_method in ['WLS', 'OLS', 'UWLLS', 'ULLS',
                                               'WLLS', 'OLLS', 'CLS', 'CWLS']
 
+        if "weights_method" in self.kwargs:
+            self.is_iter_method = True
+        else:
+            self.is_iter_method = False
+
+        if "num_iter" not in self.kwargs and self.is_iter_method:
+            self.kwargs["num_iter"] = 4
+
+        self.extra = {}
```

Comment:
self.extra is now already defined in dti.py and is able to hold a variety of items. An item inside of extra may be "reweighting_parameters", but in general extra can hold many things, such as "robust", "leverages", and so on

Reply:
I confirm for extra and thank you for the clarification @samcoveney

Reply:
(I mention extra in dti.py to point out that the implementation is already alive and working in there with this name, it would be a shame to make it less general than extra - see here #2730 )

Reply:
ok! Sorry for not noticing that! Then I am fine with maintaining that name. Our function's documentation, however, poorly mention this class attribute. Can we add some documentation on that somewhere?

Reply:
Will do!